### DISCUSSION TITLE
Programatic usage of the package

### DISCUSSION THREAD

**Discussion by renke0** [thim81/openapi-format#142] [Q&A] [answered]
I am auto-generating an openapi spec from my code, and I want to use this package to format the final document. Is there a way to import it and use the functions programatically, without having to resort to the CLI?

**Comment by thim81** [comment, marked as answer]
hi @renke0 

Yes, you can use a large part of the package functions in a NodeJS project.

We are using them in the openapi-format-playground, which is a Typescript project.
Example: https://github.com/thim81/openapi-format-playground/blob/main/src/pages/api/format.ts

TypeScript Types: https://github.com/thim81/openapi-format-playground/blob/main/types/openapi-format.d.ts

**Reply by renke0** [reply]
Thanks! This will greatly improve the current solution I have and save me several hours.

**Reply by thim81** [reply]
@renke0 Let me know, if you are missing functions not being exposed.

**Reply by renke0** [reply]
Thanks for being so supportive! 

Is it possible for you to include the [`openapi-format.d.ts`](https://github.com/thim81/openapi-format-playground/blob/main/types/openapi-format.d.ts) in the package to make it compatible with typescript?

**Reply by thim81** [reply]
Can I just take the file and drop it in Types? Or is there any advice of how to best do this?

**Reply by renke0** [reply]
Yes, dropping it in the root directory of the package should do it. You'll also need to point to the `type` file in your package json, as well as add it to the `files` array:

> `package.json`
> ```json
> {
>  "name": "openapi-format",
>  "type": "openapi-format.d.ts",
>  // remaining package json configs
>  "files": [
>    // current files already exported by the package
>    "openapi-format.d.ts"
>  ]
>}

**Reply by thim81** [reply]
@renke0 I created a [PR](https://github.com/thim81/openapi-format/pull/143), could you check it? I'm no expert in Typescript.

**Reply by thim81** [reply]
@renke0 

Can you manually put the d.ts file in your project and see if it works as expected?

**Reply by renke0** [reply]
Thanks for the PR, it looks good. 

I tested locally, everything seems to be correct.

**Reply by thim81** [reply]
hi @renke0 

We just released [openapi-format v1.24](https://github.com/thim81/openapi-format/releases/tag/v1.24.0), which includes the types.

If you have more requests/needs, let me know.
